ovos_utils/fingerprinting.py

```diff
@@ -0,0 +1,430 @@
+    get_desktop_environment, search_mycroft_core_location, is_process_running
+
+
+class MycroftPlatform(str, Enum):
+    PICROFT = "picroft"
+    BIGSCREEN = "bigscreen"
+    OVOS = "OpenVoiceOS"
+    MARK1 = "mycroft_mark_1"
+    MARK2 = "mycroft_mark_2"
+    HOLMESV = "HolmesV"
+    OLD_HOLMES = "mycroft-lib"
+    NEON = "neon_core"
+    CHATTERBOX = "chatterbox"
+    OTHER = "unknown"
+
+
+def detect_platform():
+    return max(((k, v) for k, v in classify_fingerprint().items()),
+               key=lambda k: k[1])[0]
+
+
+def get_config_fingerprint(config=None):
+    from ovos_utils.configuration import read_mycroft_config
+    conf = config or read_mycroft_config()
+    listener_conf = conf.get("listener", {})
+    skills_conf = conf.get("skills", {})
+    return {
+        "enclosure": conf.get("enclosure", {}).get("platform"),
+        "data_dir": conf.get("data_dir"),
+        "msm_skills_dir": skills_conf.get("msm", {}).get("directory"),
+        "ipc_path": conf.get("ipc_path"),
+        "input_device_name": listener_conf.get("device_name"),
+        "input_device_index": listener_conf.get("device_index"),
+        "default_audio_backend": conf.get("Audio", {}).get("default-backend"),
+        "priority_skills": skills_conf.get("priority_skills"),
+        "backend_url": conf.get("server", {}).get("url")
+    }
+
+
+def get_platform_fingerprint():
+    return {
+        "hostname": socket.gethostname(),
+        "platform": platform.platform(),
+        "python_version": platform.python_version(),
+        "system": platform.system(),
+        "version": platform.version(),
+        "arch": platform.machine(),
+        "release": platform.release(),
+        "desktop_env": get_desktop_environment(),
+        "mycroft_core_location": search_mycroft_core_location(),
+        "can_display": has_screen(),
+        "is_gui_installed": is_installed("mycroft-gui-app"),
+        "is_vlc_installed": is_installed("vlc"),
+        "pulseaudio_running": is_process_running("pulseaudio"),
+        "core_supports_xdg": core_supports_xdg(),
+        "core_version": {
+            "version_str": get_mycroft_version(),
+            "is_chatterbox_core": is_chatterbox_core(),
+            "is_neon_core": is_neon_core(),
+            "is_holmes": is_holmes(),
+            "is_ovos": is_ovos(),
+            "is_mycroft_core": is_mycroft_core()
+        }
+    }
+
+
+def get_fingerprint():
+    finger = get_platform_fingerprint()
+    finger["configuration"] = get_config_fingerprint()
+    return finger
+
+
+def core_supports_xdg():
+    if any((is_holmes(), is_ovos(), is_neon_core(), is_chatterbox_core())):
+        return True
+    # mycroft-core does not support XDG as of 10 may 2021
+    # however there are patched versions out there, eg, alpine package
+    # check if the .conf exists in new location
+    # TODO deprecate
+    return isfile(join(XDG.save_config_path('mycroft'), 'mycroft.conf'))
+
+
+def get_mycroft_version():
```

Comment:
Is this intended to work for forks as well? If yes, might be worth trying to extract from setup.py

Reply:
setup.py doesnt get copied over when pip installed, but it would work for git installs, can you think of any case where this is in setup.py but not in the version file?

setup.py seems to read this string from version file and not easy to parse

Reply:
feels like setup.py is a good addition, but not here, instead a new method to get package_name?